File: alibabacloud-gateway-sls/php/tests/signature_version_test.php
<?php
require __DIR__ . '/../vendor/autoload.php';

use Darabonba\GatewaySls\Client;
use Darabonba\GatewaySpi\Models\InterceptorContext;
use Darabonba\GatewaySpi\Models\InterceptorContext\request;
use Darabonba\GatewaySpi\Models\InterceptorContext\configuration;

$cases = json_decode(file_get_contents(__DIR__ . '/../../testdata/signature-version.json'), true);
$client = new Client();
foreach ($cases as $case) {
    $context = new InterceptorContext([
        'request' => new request(['signatureVersion' => $case['version']]),
        'configuration' => new configuration(['regionId' => $case['region'], 'endpoint' => $case['endpoint']])
    ]);
    $version = $client->getSignatureVersion($context);
    if ($version !== $case['want'] || $context->configuration->regionId !== $case['resolved'] || $context->request->signatureVersion !== $case['version']) {
        throw new \RuntimeException('Failed signature version case: ' . $case['name']);
    }
}
echo count($cases) . " signature version cases passed\n";

File: alibabacloud-gateway-sls/csharp/tests/SignatureVersionTest.cs
using System;
using System.IO;
using System.Threading.Tasks;
using AlibabaCloud.GatewaySls;
using AlibabaCloud.GatewaySpi.Models;
using Newtonsoft.Json.Linq;
using Xunit;

namespace tests
{
    public class SignatureVersionTest
    {
        [Fact]
        public async Task SelectsSignatureVersion()
        {
            var cases = JArray.Parse(File.ReadAllText(Path.Combine(AppContext.BaseDirectory, "signature-version.json")));
            var client = new Client();
            foreach (var test in cases)
            {
                foreach (var asynchronous in new[] { false, true })
                {
                    var context = new InterceptorContext
                    {
                        Request = new InterceptorContext.InterceptorContextRequest { SignatureVersion = (string)test["version"] },
                        Configuration = new InterceptorContext.InterceptorContextConfiguration
                        {
                            RegionId = (string)test["region"],
                            Endpoint = (string)test["endpoint"]
                        }
                    };
                    var version = asynchronous ? await client.GetSignatureVersionAsync(context) : client.GetSignatureVersion(context);
                    Assert.True(version == (string)test["want"], (string)test["name"]);
                    Assert.Equal((string)test["resolved"], context.Configuration.RegionId);
                    Assert.Equal((string)test["version"], context.Request.SignatureVersion);
                }
            }
        }
    }
}
